components/src/app/local-storage.service.ts

```diff
@@ -0,0 +1,29 @@
+import { FactoryProvider, InjectionToken, PLATFORM_ID } from '@angular/core';
+import { isPlatformBrowser } from '@angular/common';
+
+export const LOCAL_STORAGE = new InjectionToken('LocalStorageToken');
+
+export function localStorageFactory(platformId: Object): Storage {
```

Comment:
We do always return `Storage` which means that from now on we do not forced to wrap all calls to localStorage in if statements